File: 3.3P/Clock/Clock/Clock.cs
using System;

namespace Clock
{
    public class Clock
    {
        private Counter hour;
        private Counter minute;
        private Counter second;

        public Clock()
        {
            hour = new Counter("Hour");
            minute = new Counter("Minute");
            second = new Counter("Second");
        }

        public void Tick()
        {
            if (second.Tick == 59)
            {
                second.Reset();
                if (minute.Tick == 59)
                {
                    minute.Reset();
                    if (hour.Tick == 23)
                    {
                        hour.Reset();
                        minute.Reset();
                        second.Reset();
                    }
                    else
                    {
                        hour.Increment();
                    }
                }
                else
                {
                    minute.Increment();
                }
            }
            else
            {
                second.Increment();
            }
        }

        public void Reset()
        {
            hour.Reset();
            minute.Reset();
            second.Reset();
        }

        public string time
        {
            get 
            {
                return $"{hour.Tick:00}:{minute.Tick:00}:{second.Tick:00}";
            }
        }
    }
}

File: 11.1P/clock.py
from counter import Counter


class Clock:
    def __init__(self):
        self.hours = Counter("hours")
        self.minutes = Counter("minutes")
        self.seconds = Counter("seconds")

    def Tick(self):
        self.seconds.Increment()
        if self.seconds.Tick > 59:
            self.minutes.Increment()
            self.seconds.Reset()
            if self.minutes.Tick > 59:
                self.hours.Increment()
                self.minutes.Reset()
                if self.hours.Tick > 23:
                    self.Reset()

    def Reset(self):
        self.hours.Reset()
        self.minutes.Reset()
        self.seconds.Reset()

    @property
    def Time(self):
        return str(self.hours.Tick).zfill(2) + ":" + str(self.minutes.Tick).zfill(2) + ":" + str(
            self.seconds.Tick).zfill(2)